reuse.go

```diff
@@ -0,0 +1,187 @@
+package libp2pquic
+
+import (
+	"errors"
+	"net"
+	"sync"
+
+	srcs "github.com/lnykww/go-src-select"
+)
+
+type reuseConn struct {
+	net.PacketConn
+	mutex sync.Mutex
+	ref   int
+}
+
+func NewReuseConn(conn net.PacketConn) *reuseConn {
+	return &reuseConn{
+		PacketConn: conn,
+		ref:        1,
+	}
+}
+
+func (rc *reuseConn) Ref() bool {
```

Comment:
Things will become a lot clearer if you introduce a function to tell if `rc.ref > 0` and another function to increase the value.